### PR TITLE
Revert .yamllint to match lineage; fix issues flagged by yamllint.

### LINKED ISSUE #26: Resolve yamllint Issues
# 🐛 Summary #
Revert .yamllint changes made in PR #5 and fix all issues flagged by pre-commit.
To run pre-commit locally,  run `pre-commit yamllint --all-files`

To setup pre-commit, run `./setup-env`.
If it fails to run, follow the instructions returned in the console. This includes installing pyenv pyenv-virtualenv, and adding its path to .zhrc.

To run the check-case-conflict hook, run `pre-commit run check-case-conflict --all-files`.
To run a different hook, replace `check-case-conflict` with the name of the hook. The list of available hooks is in .pre-commit-config.yaml

### PR DESCRIPTION
**Pull request by Matthew-Grayson**
#  #

## 🗣 Description ##
- Remove duplicate github-actions entry and commented out code from dependabot.yml
- Revert .yamllint changes made in PR #5
- Fix all issues flagged by the pre-commit hook, yamllint
  - Fix lines that are too long
  - Add `---` to beginning of all *.yml files
  - Remove extra spaces in brackets
- Refactor yml files for consistency.
  - Remove quotes from strings unless needed
  - Convert double quotes to single quotes when possible (single quotes don't process escape characters like "\n")
- Convert .eslintrc files to pure yaml
- Remove steps from .github/workflows/build.yml related to Packer and Go formatting.

## 💭 Motivation and context ##
Closes issue #26 

## 🧪 Testing ##
All  jest tests pass.
Running `npx eslint --print-config .eslintrc.yml` before and after my changes produces the same output.

Run pre-commit locally to test: `pre-commit yamllint --all-files`

To setup pre-commit, run ./setup-env.
If it fails to run, follow the instructions returned in the console. This includes installing pyenv pyenv-virtualenv, and adding its path to .zhrc.

To run the check-case-conflict hook, run pre-commit run check-case-conflict --all-files.
To run a different hook, replace check-case-conflict with the name of the hook. The list of available hooks is in .pre-commit-config.yaml

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [ ] Tests have been added and/or modified to cover the changes in this PR.
- [ ] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Create a release.

### CODE REVIEW

**Review by cduhn17 — APPROVED** [review]
LGTM

**Comment by dav3r** [comment]
What would it take to get to the point where this PR shows the GitHub "build / lint" action passing the `yamllint` pre-commit check?  I'd be ok with other pre-commit checks failing still (presumably those will be tackled in other PRs), as long as we can see pre-commit passing the thing you are fixing in this PR (the `yamllint` check).

**Comment by Matthew-Grayson** [comment]
> What would it take to get to the point where this PR shows the GitHub "build / lint" action passing the `yamllint` pre-commit check? I'd be ok with other pre-commit checks failing still (presumably those will be tackled in other PRs), as long as we can see pre-commit passing the thing you are fixing in this PR (the `yamllint` check).

Done! The issue was, I removed the quotes around  `go-version: '1.20'`.
`go-version: 1.20` was being parsed as version 1.2 which caused additional issues down the line.
Additionally, I combed the rest of the *.yml files to ensure all version numbers are enclosed in single quotes (with some exceptions where a number is required).

And you're right: I'm tackling the rest of the lint issues in separate PRs

**Review by schmelz21 — APPROVED** [review]

File: .github/dependabot.yml
Comment (dav3r) [inline .github/dependabot.yml:17]: Any upstream dependencies should be handled by the parent repo (e.g. skeleton-generic, in this case).  For an example of how we typically do that, please see [here](https://github.com/cisagov/pshtt/blob/17d7598993053d30e03455d30baf03b77392d74e/.github/dependabot.yml#L3-L30).

**Comment by Matthew-Grayson** [comment]
@dav3r, would you please review this. It needs approval from you or one of the other .github/ codeowners.

**Review by dav3r — CHANGES_REQUESTED** [review]
Please see the one item that I noted.

File: .github/dependabot.yml
Comment (Matthew-Grayson) [reply .github/dependabot.yml:17]: This will look a bit different because I refactored to match the existing dependabot.yml code from cisagov/crossfeed.

Comment (dav3r) [reply .github/dependabot.yml:17]: For the future, we let the person who opened the conversation be the one to resolve it.  Also, it's helpful to include the commit where you addressed the concern in the convo (in this case it was https://github.com/cisagov/XFD/pull/32/commits/adb7284c9742ebd1ed18c7274b2b434e0daf2310).  This looks good to me, so we can leave this convo as resolved.

**Review by dav3r — APPROVED** [review]
Approved since [yamllint passes pre-commit now](https://github.com/cisagov/XFD/actions/runs/8269484240/job/22624969479?pr=32#step:15:213) and @Matthew-Grayson stated that the [remaining automated test failures will be addressed in future PRs](https://github.com/cisagov/XFD/pull/32#issuecomment-1994888253).

**Pull request merged**
